packages/p2p/clients/js/index.js

```diff
@@ -0,0 +1,208 @@
+const WebSocket = require('rpc-websockets').Client
+
+/*** STATE ***/
+
+let server
+let ourId
+let theirId
+let room
+let events = {}
+let survivingPeer
+let offeringPeer
+let answeringPeer
+let dataChannel
+let connectionStatusCB
+let messageCB
+
+/*** MAIN FUNCTIONS ***/
+
+function connect(userId, uri) {
+  // Set user ID
+  ourId = userId
+
+  server = null
+  theirId = null
+  room = 'empty'
+  survivingPeer = null
+  offeringPeer = null
+  answeringPeer = null
+  dataChannel = null
+  connectionStatusCB = null
+
+  // Connect to server
+  server = new WebSocket(uri)
+
+  // Set up offering peer
+  offeringPeer = new RTCPeerConnection()
+
+  // Set up offering peer's data channel
+  dataChannel = offeringPeer.createDataChannel("dataChannel")
+
+  // Set up answering peer
+  answeringPeer = new RTCPeerConnection()
+
+  // Allow answering peer to accept a data channel from an offer
+  answeringPeer.ondatachannel = handleReceiveDataChannel
+}
+
+function disconnect() {
+  dataChannel.close()
+  leaveRoom()
+  server = null
+  theirId = null
+  events = {}
+  messageCB = null
+}
+
+function listenConnectionStatus(callback) {
+  connectionStatusCB = callback
+  dataChannel.onopen = callback
+  dataChannel.onclose = callback
+}
+
+function listenMessage(callback) {
+  messageCB = callback
+  dataChannel.onmessage = callback
+}
+
+function whenServerReady(callback) {
+  server.on('open', callback)
+}
+
+function whenServerDisconnect(callback) {
+  server.on('close', callback)
+}
+
+function eventName(type, key) {
+  return `${type}-${key}`
+}
+
+function joinRoom(roomName) {
+  room = roomName
+  server.call('joinRoom', [room, ourId])
+}
+
+function leaveRoom() {
+  server.call('leaveRoom', [room, ourId])
+  room = 'empty'
+}
+
+async function rollCall() {
+  const peers = await server.call('rollCall', [room])
+  return peers.filter(id => id != ourId)
+}
+
+function listenOffers(callback) {
+  listenEvent('offer', callback)
+  listenEvent('answer', (e) => {
+    acceptAnswer(e.from, e.data)
+  })
+  listenEvent('candidate', (e) => {
+    addCandidate(e.from, e.data)
+  })
+}
+
+function listenEvent(eventType, callback) {
```

Comment:
Is there a reason we are renaming `subscribe` and `uncubscribe` to `listen` and `stopListen`. If we can we should try to use the established language